### ISSUE TITLE
TODO: sflib.py: why another except block?

### ISSUE THREAD

**Issue by bcoles** [smicallef/spiderfoot#887]
```python
                # TODO: why another except block?
                try:
                    self.error(f"Unexpected exception ({e}) occurred fetching URL: {url}", False)
                    self.error(traceback.format_exc(), False)
                except BaseException as f:
                    # TODO: why is this exception ignored?
                    self.debug(f"Ignoring exception: {f}")
                    return result
```

**Comment by leotrubach** [comment]
`except BaseException` is not best idea... It captures Ctrl+C also

**Comment by bcoles** [comment]
This was "fixed"in #1076 by grouping functionality together in several `try / except` blocks, although the `fetchUrl()` function is still a mess.

> 
> 
> `except BaseException` is not best idea... It captures Ctrl+C also

Fixed
